api_end_end_testing/src/test/java/tds/user/api/tests/UserApiTests.java

```diff
@@ -0,0 +1,359 @@
+     * Test of Delete User, HTTP DELETE of /rest/external/user, 204 success item found and deleted
+     */
+    @Test
+    public void shouldCreateAndUpdateUserToMultipleRoles() {
+        String randomUserEmail = createRandomUserEmail();
+
+        UserInfo userInfo = createUserOneRoleAssoc(randomUserEmail);
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "32467"));
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "11276"));
+
+        // Update the user with new information
+        userInfo.setFirstName("Miranda");
+        userInfo.setLastName("Bailey");
+        userInfo.setPhoneNumber("415-332-9090");
+        userInfo.setRoleAssociations(roleAssociations);
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+        .when()
+            .post(uriLocation)
+        .then()
+            .statusCode(204);
+
+        // Execute a DELETE to delete the user
+        deleteExistingUser(randomUserEmail);
+
+        // Execute a GET by email to validate user is deleted
+        given()
+            .contentType(ContentType.JSON)
+            .accept(ContentType.JSON)
+            .header(authHeader)
+            .when()
+            .get(uriLocation + "/" + randomUserEmail + "/details")
+            .then()
+            .statusCode(404);
+    }
+
+    /*
+     * Test of Create User with invalid email, HTTP POST of /rest/external/user, 400 bad request
+     */
+    @Test
+    public void shouldNotCreateUserWithInvalidEmail() {
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "19037"));
+
+        UserInfo userInfo = new UserInfo("bademail", "Judy", "Bloom", "808-443-1199", roleAssociations);
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+            .when()
+            .post(uriLocation)
+            .then()
+            .statusCode(400)
+            .body("messages.email[0]", equalTo("User Email Address must be a valid email address"));
+    }
+
+    /*
+     * Test of Create User with invalid first name, HTTP POST of /rest/external/user, 400 bad request
+     */
+    @Test
+    public void shouldNotCreateUserWithInvalidFirstName() {
+        String randomUserEmail = createRandomUserEmail();
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "55342"));
+
+        UserInfo userInfo = new UserInfo(randomUserEmail, "", "Dodge", "714-228-4848", roleAssociations);
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+            .when()
```

Comment:
It may be a setting you have configured in IntelliJ.  Prior to commit there are a bunch of checkboxes on the right.  If the format code one is checked then it'll do this to you on commit.  Or if you hit option + command + enter.

It isn't ideal.